### DISCUSSION TITLE
How do you add text to a Handle?

### DISCUSSION THREAD

**Discussion by jtwigg** [xyflow/xyflow#2095] [Q&A]
I'm having to make text in the my custom views, visually align them with input/output nodes. 

While this works in the current screenshot, its a little off, and if my inputs/outputs start to have dynamic heights, it will be impossible to work.

![image](https://user-images.githubusercontent.com/1237233/164277144-3ff003a5-ebb1-403e-9916-f7ea70b301d9.png)

**Comment by rohansh-tty** [comment]
I'm having the same problem, I need to name my node handles. Is there any workaround for this @moklick?

**Reply by jtwigg** [reply]
My particular solution, unfortunately was to migrate to a different library; @projectstorm/react-diagrams.

This [reddit thread](https://www.reddit.com/r/reactjs/comments/u894mi/looking_for_node_editor_gui_logic_block_editor/) spawned as lot of conversation about the variety of  JS/TS `node editors` available, including feedback from the owner of **this** `react-flow` repo.

My reasons for moving to `react-diagrams` were
* I was able to customize all of my Nodes, Edges, "Ports" and React UI components heavily.
* it uses a fairly straight forward MVC model that makes it very obvious how to create and override these components.

The docs are a 5/10 but the examples were sufficient to figure out how to customize what I described.

**Comment by moklick** [comment]
Hey @jtwigg ! There are multiple ways to achieve this. The easiest way would be to pass a child node to the Handle component. I created an example for this: https://codesandbox.io/s/react-flow-custom-node-dprfr0?file=/src/CustomNode.tsx:193-329  A handle is just a div. You could also wrap it with another container component and layout the corresponding components accordingly.

**Reply by guyguy2001** [reply]
Thanks for the help :)

I'd like to note, mainly for my own sanity, that you can see an example of this in the features overview example - https://reactflow.dev/docs/examples/overview/
Perhaps I just missed this, but I couldn't find any dedicated example; I think it's an important feature, that deserves a prominent example
